### PR TITLE
[MX-189] - Fix for auctions pointy modal corner and non-fullscreen display

### PR DESCRIPTION
**Pull request by brainbicycle**
https://artsyproduct.atlassian.net/browse/MX-189

- Fixes pointy corner of modals in auctions flows
- Makes live auctions views presentation full screen on iPad + iOS 13

While we were working on the pointy corner Damon noticed that the LiveAuctions VC was not being presented correctly on iPad, this is due to the new card presentation style for modals in iOS 13. Fix is to add an exception for this view controller in our presentation code. There may be more layout issues associated with our native modals on iOS 13 that might be worth looking into. 

**Co-authored with @dzucconi** 

**Before**
pointyModal

notFullScreenLiveAuction

**After**
fixedPointyModal

fullScreenLiveAuction

### CODE REVIEW

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.h [hunk 1]
```diff
@@ -72,8 +72,6 @@ NS_ASSUME_NONNULL_BEGIN
 /// Not all view controllers should be presented as a push, use this to determine whether the topVC will present modally or push.
 + (BOOL)shouldPresentViewControllerAsModal:(UIViewController *)viewController;
 
-- (BOOL)isShowingStatusBar;
```
Comment (brainbicycle) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.h:75]: This method was removed recently, removed the declaration in header which was causing a warning

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m [hunk 2]
```diff
@@ -478,7 +488,9 @@ - (void)pushViewController:(UIViewController *)viewController animated:(BOOL)ani
             } else if ([viewController isKindOfClass:UINavigationController.class] && [[(UINavigationController *)viewController topViewController] isKindOfClass:ARShowConsignmentsFlowViewController.class]) {
                 // Consignments gets full screen
                 viewController.modalPresentationStyle = UIModalPresentationFullScreen;
-            }
+            } else if ([self.class shouldPresentModalFullScreen:viewController]) {
+                viewController.modalPresentationStyle = UIModalPresentationFullScreen;
+            } else {}
```
Comment (ds300) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m:493]: nonblocking

```suggestion
            }
```

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m [hunk 3]
```diff
@@ -460,6 +460,16 @@ + (BOOL)shouldPresentViewControllerAsModal:(UIViewController *)viewController
...
 
++ (BOOL)shouldPresentModalFullScreen:(UIViewController *)viewController {
+    NSArray *fullScreenClasses = @[LiveAuctionViewController.class];
+    for (Class klass in fullScreenClasses) {
+        if ([viewController isKindOfClass:klass]) {
+            return YES;
+        }
+    }
+    return NO;
+}
```
Comment (ds300) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m:471]: nonblocking: do we expect this list to get bigger? the method body could be just `return [viewController isKindOfClass:LiveAuctionViewController.class]` for now at least.

Comment (brainbicycle) [reply Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m:471]: Yeah this is probably an overly complicated implementation, I will update 👍

**Pull request merged**